docs/distributed/distributed_autofaiss.md

```diff
@@ -0,0 +1,222 @@
+Make sure to open your spark UI, at http://localhost:8080 (or the ip where the master node is running)
+
+Save this script to indexing.py.
+
+Then run `./autofaiss.pex indexing.py`
+
+```python
+from autofaiss import build_index
+from pyspark.sql import SparkSession  # pylint: disable=import-outside-toplevel
+
+from pyspark import SparkConf, SparkContext
+
+def create_spark_session():
+    # this must be a path that is available on all worker nodes
+    
+    os.environ['PYSPARK_PYTHON'] = "/home/ubuntu/autofaiss.pex"
+    spark = (
+        SparkSession.builder
+        .config("spark.submit.deployMode", "client") \
+        .config("spark.executorEnv.PEX_ROOT", "./.pex")
+        .config("spark.executorEnv.PYTHONUTF8", "1")
+        #.config("spark.executor.cores", "16")
+        #.config("spark.cores.max", "48") # you can reduce this number if you want to use only some cores ; if you're using yarn the option name is different, check spark doc
+        .config("spark.task.cpus", "16")
+        .config("spark.driver.port", "5678")
+        .config("spark.driver.blockManager.port", "6678")
+        .config("spark.driver.host", "172.31.35.188")
+        .config("spark.driver.bindAddress", "172.31.35.188")
+        .config("spark.executor.memory", "18G") # make sure to increase this if you're using more cores per executor
+        .config("spark.executor.memoryOverhead", "8G")
+        .config("spark.task.maxFailures", "100")
+        .master("spark://172.31.35.188:7077") # this should point to your master node, if using the tunnelling version, keep this to localhost
+        .appName("spark-stats")
+        .getOrCreate()
+    )
+    return spark
+
+spark = create_spark_session()
+
+index, index_infos = build_index(
+    embeddings="hdfs://root/path/to/your/embeddings/folder",
+    distributed="pyspark",
+    file_format="parquet",
+    max_index_memory_usage="16G",
+    current_memory_available="24G",
+    temporary_indices_folder="hdfs://root/tmp/distributed_autofaiss_indices",
+    index_path="hdfs://root/path/to/your/index/knn.index",
+    index_infos_path="hdfs://root/path/to/your/index/infos.json"
+)
+
+```
+
+Another example:
+
+```python
+index, index_infos = build_index(
+    embeddings=["s3://laion-us-east-1/embeddings/vit-l-14/laion2B-en/img_emb","s3://laion-us-east-1/embeddings/vit-l-14/laion2B-multi/img_emb","s3://laion-us-east-1/embeddings/vit-l-14/laion1B-nolang/img_emb"],
+    distributed="pyspark",
+    max_index_memory_usage="200G",
+    current_memory_available="24G",
+    nb_indices_to_keep=10,
+    file_format="npy",
+    temporary_indices_folder="s3://laion-us-east-1/mytest/my_tmp_folder5",
+    index_path="s3://laion-us-east-1/indices/vit-l-14/image/knn.index",
+    index_infos_path="s3://laion-us-east-1/indices/vit-l-14/image/infos.json"
+)
+```
+
+## Benchmark
+
+Computing a 168GB multi pieces index `OPQ24_168,IVF131072_HNSW32,PQ24x8` index on 5550336490 embeddings of dim 768 using 10 nodes with 16 cores (c6i.4xlarge) 
```

Comment:
index `OPQ24_168,IVF131072_HNSW32,PQ24x8` index -> index `OPQ24_168,IVF131072_HNSW32,PQ24x8` ?
(maybe the latter index is duplicated)

Reply:
fixed